include/wsrep/server_service.hpp

```diff
@@ -190,6 +189,22 @@ namespace wsrep
          */
         virtual void debug_sync(const char* sync_point) = 0;
 
+        /**
+         * Set encryption key
+         */
+        virtual int set_encryption_key(const unsigned char* key, 
+                                       int key_length) = 0;
```

Comment:
As I mentioned in the https://github.com/codership/mariaDB-wsrep/pull/43 `set_encryption_key()` is a call FROM the server straight to provider and it does not seem to need to involve anything server-specific. It is likely unnecessary in the `wsrep::server_service interface`.

It better be moved to `wsrep::server_state` interface - as it is its job is to abstract provider functionality.

Reply:
Commented on mariadb-wsrep about reasons why server_service is chosen instead of server_state.

Reply:
Added set_encryption_key to be member of server_state interface.